ui/jetbrains/src/main/java/com/neueda/jetbrains/plugin/graphdb/jetbrains/util/TaskExecutor.java

```diff
@@ -0,0 +1,44 @@
+package com.neueda.jetbrains.plugin.graphdb.jetbrains.util;
+
+import com.intellij.openapi.application.ApplicationManager;
+
+import java.util.concurrent.Callable;
+import java.util.function.Consumer;
+
+public class TaskExecutor implements Executor {
+
+    public static final String SYNC_TASK_EXECUTION = "syncTaskExecution";
+
+    public static Executor getInstance() {
+        return ExecutorHolder.INSTANCE;
+    }
+
+    public <T> void runInBackground(Callable<T> task, Consumer<T> onSuccess, Consumer<Exception> onFailure) {
+        ApplicationManager.getApplication().executeOnPooledThread(() -> {
+            try {
+                T result = task.call();
+                ApplicationManager.getApplication().invokeLater(() -> onSuccess.accept(result));
+
+            } catch (Exception exception) {
+                ApplicationManager.getApplication().invokeLater(() -> onFailure.accept(exception));
+            }
+        });
+    }
+
+    private static class SyncTaskExecutor implements Executor {
+        public <T> void runInBackground(Callable<T> task, Consumer<T> onSuccess, Consumer<Exception> onFailure) {
+            try {
+                T result = task.call();
+                onSuccess.accept(result);
+            } catch (Exception e) {
+                onFailure.accept(e);
+            }
+        }
+    }
+
+    private static class ExecutorHolder {
+        static final Executor INSTANCE =
+                System.getProperty(SYNC_TASK_EXECUTION) == null ? new TaskExecutor() : new SyncTaskExecutor();
```

Comment:
Please do not use such hacks in application. If you register `TaskExecutor` as component or service, you will be able to override implementation for tests:

```java
public void setUp()
//...
((MutablePicoContainer) getProject().getPicoContainer()).registerComponentInstance(MyInt.class, new MyTestImpl());
```
See:

* https://www.jetbrains.org/intellij/sdk/docs/basics/plugin_structure/plugin_components.html
* https://www.jetbrains.org/intellij/sdk/docs/basics/plugin_structure/plugin_services.html